woocommerce/woocommerce-android
# Issue/4754 Connection status not in sync between 2 sources

Closes: #4754 


### Description
I found out that TerminalListenerImpl::onConnectionStatusChange invoked with Connected right after we connected to a reader, but before the update check started.  Callback ReaderCallback we pass to the connectToReader method returns only AFTER the required update has been finished.

We use readerStatus exposed by TerminalListenerImpl to show the Connected/NotConnected state of the details screen. When we are connecting to a reader from the settings, you can notice that the "Connected" state is shown, and only after sometime Connection dialog is dismissed. It's part of the problem, the bigger issue is that if the user clicks the "cancel" button on the dialog after the Connected state emitted in TerminalListenerImpl but the connection (including required update check and update) itself is not finished then we just dismiss the Connection dialog and showing "Connected" stated which messes things up, because the update can still be running in the background and we can not do anything with the reader.

The PR makes `readerStatus` single source of truth. I manually update it's state from "ConnectionManager"

### Testing instructions
Connecting:
* Settings -> In Person Payments -> Manage Readers -> Connect
* Notice that during connection the details screen stays in "Not Connected" state

Payment:
* Try to collect payment. Notice that there is no regression


### Images/gif


https://user-images.githubusercontent.com/4923871/132650167-bd8d3623-2399-43c0-a8e7-d053ebc9332a.mp4
